lib/manageiq/api/common/application_controller_mixins/parameters.rb

```diff
@@ -0,0 +1,95 @@
+module ManageIQ
+  module API
+    module Common
+      module ApplicationControllerMixins
+        module Parameters
+          def params_for_create
+            # We already validate this with OpenAPI validator, that validates every request, so we shouldn't do it again here.
+            body_params.permit!
+          end
+
+          def safe_params_for_list
+            # :limit & :offset can be passed in for pagination purposes, but shouldn't show up as params for filtering purposes
+            @safe_params_for_list ||= params.merge(params_for_polymorphic_subcollection).permit(*permitted_params, :filter => {})
+          end
+
+          def permitted_params
+            api_doc_definition.all_attributes + [:limit, :offset] + [subcollection_foreign_key]
+          end
+
+          def subcollection_foreign_key
+            "#{request_path_parts["primary_collection_name"].singularize}_id"
```

Comment:
I know it's not from this PR, but we a more Rails oriented way to do this...there's should be a helper that gets you this string